### PR TITLE
Edits to Custom Scalars article

### PR DESCRIPTION
**Pull request by StephenBarlow**
Also moves enum content into the Schema Basics article

### CODE REVIEW

File: docs/source/schema/custom-scalars.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+```
+
+## Example: Restricting integers to odd values
+
+In this example, we create a custom scalar called `Odd` that can only contain odd integers:
+
+```js{20-31}
+const { ApolloServer, gql } = require('apollo-server');
+const { GraphQLScalarType } = require('graphql');
+const { Kind } = require('graphql/language');
```
Comment (trevor-scheer) [inline docs/source/schema/custom-scalars.md:124]:
```suggestion
const { GraphQLScalarType, Kind } = require('graphql');
```

File: docs/source/schema/custom-scalars.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  typeDefs,
+  resolvers
+});
+```
+
+## Example: Restricting integers to odd values
+
+In this example, we create a custom scalar called `Odd` that can only contain odd integers:
+
+```js{20-31}
```
Comment (trevor-scheer) [inline docs/source/schema/custom-scalars.md:121]: If you take my suggestion below, I believe the highlighted lines will need to be decremented by 1, but please verify my assumption.
```suggestion
```js{19-30}
```

File: docs/source/schema/custom-scalars.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+}
+
+const resolvers = {
+  Odd: new GraphQLScalarType({
+    name: 'Odd',
+    description: 'Odd custom scalar type',
+    parseValue: oddValue,
+    serialize: oddValue,
+    parseLiteral(ast) {
+      if (ast.kind === Kind.INT) {
```
Comment (trevor-scheer) [inline docs/source/schema/custom-scalars.md:147]: Nice catch on the previously incorrect `Kind.Int` when transposing / recreating this example!

**Review by trevor-scheer — APPROVED** [review]
Wonderful as always 🎉

File: docs/source/schema/custom-scalars.md (hunk 3 above)
Comment (StephenBarlow) [reply docs/source/schema/custom-scalars.md:147]: Ha I *wish* I could take credit for this but I think it was already correct 😄

**Pull request merged**